### PR TITLE
PLANET-1943 Initial version of submenu block

### PR DESCRIPTION
**Pull request by kirdia**
[Issue PLANET-1943](https://jira.greenpeace.org/browse/PLANET-1943)

### CODE REVIEW

File: classes/controller/blocks/class-submenu-controller.php [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+			$link1    = $attributes['link1'] ?? 'false';
+			$link2    = $attributes['link2'] ?? 'false';
+
+			$menu = $this->parse_post_content( $content, [ $heading1, $heading2 ], [ $link1, $link2 ] );
+
+			wp_enqueue_script( 'submenu', P4BKS_ADMIN_DIR . 'js/submenu.js' );
+			wp_localize_script(
+				'submenu',
+				'submenu',
+				wp_json_encode( $menu )
```
Comment (angtheod) [inline classes/controller/blocks/class-submenu-controller.php]: 3rd arg of wp_localize_script should be array but wp_json_encode returns false/string.
So, we should put it in an array like
`[ wp_json_encode( $menu ) ]`

File: classes/controller/blocks/class-submenu-controller.php [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+				}
+			}
+
+			return $menu;
+		}
+
+		/**
+		 * Create a std object representing a node/heading.
+		 *
+		 * @param \DOMElement $node
```
Comment (angtheod) [inline classes/controller/blocks/class-submenu-controller.php]: missing doc comments

File: classes/controller/blocks/class-submenu-controller.php [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+			$heading              = intval( $heading );
+			if ( ! empty( $heading ) &&
+				in_array( $heading, $valid_heading_values, true ) ) {
+				return true;
+			}
+
+			return false;
+		}
+	}
+}
```
Comment (angtheod) [inline classes/controller/blocks/class-submenu-controller.php]: please add a new line at the end of the file

File: admin/templates/submenu.tpl.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
+<script type="text/html" id="tmpl-shortcode-ui-field-p4-radio">
```
Comment (angtheod) [inline admin/templates/submenu.tpl.php]: Does this file need to be a .php? Codesniffer complains that it did not find any php code or php short open tags

File: classes/controller/blocks/class-submenu-controller.php [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+						],
+						[
+							'value' => '2',
+							'label' => __( 'Short full-width', 'planet4-blocks' ),
+							'desc'  => 'Use: on long pages (more than 5 screens) when list items are short (up to 10 words)<br>Max items<br>recommended:9<br>Example',
+						],
+						[
+							'value' => '3',
+							'label' => __( 'Short sidebar', 'planet4-blocks' ),
+							'desc'  => 'Use: on long pages (more than 5 screens) when list items are short (up to 10 words)<br>Max items<br>recommended:9<br>Example',
```
Comment (angtheod) [inline classes/controller/blocks/class-submenu-controller.php]: Perhaps different descriptions for the 3 radio buttons have not been provided by the design, but just mentioning this

**Comment by angtheod** [comment]
Overall looks good, just some minor comments.

File: classes/controller/blocks/class-submenu-controller.php [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+		 *
+		 * @param string $template Template name.
+		 *
+		 * @return bool|string
+		 */
+		private function get_template( $template ) {
+
+			$template = P4BKS_PLUGIN_DIR . '/admin/templates/' . $template . '.tpl.php';
+			if ( file_exists( $template ) ) {
+				$contents = file_get_contents( $template );
```
Comment (angtheod) [inline classes/controller/blocks/class-submenu-controller.php]: if possible replace file_get_contents with wp_remote_get as WP CS suggest

File: classes/controller/blocks/class-submenu-controller.php [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+				'listItemImage' => '<img src="' . esc_url( plugins_url() . '/planet4-plugin-blocks/admin/images/submenu.png' ) . '" />',
+				'attrs'         => $fields,
+				'post_type'     => P4BKS_ALLOWED_PAGETYPE,
+			];
+
+			shortcode_ui_register_for_shortcode( 'shortcake_' . self::BLOCK_NAME, $shortcode_ui_args );
+		}
+
+		/**
+		 * Callback for content four column shortcode.
```
Comment (sagarsdeshmukh) [inline classes/controller/blocks/class-submenu-controller.php]: Doc block description needs to change.

File: admin/templates/submenu.tpl.php (hunk 4 above)
Comment (kirdia) [reply admin/templates/submenu.tpl.php]: Actually no, it does not have to be a php file. I just followed shortcake's convention, which names underscore templates with the extension .tpl.php. This file contains html with underscore commands and all these templates are appended at the end of the dom, to be used from backbone views objects.

File: classes/controller/blocks/class-submenu-controller.php (hunk 6 above)
Comment (kirdia) [reply classes/controller/blocks/class-submenu-controller.php]: Codesniffer locally suggests using WP_Filesystem, but for some reason which I don't remember I did not use it. I'll try it again.
No need to use wp_remote_get, it is a local file.

**Pull request merged**